helm/prefect-server/templates/agent/deployment.yaml

```diff
@@ -36,7 +36,8 @@ spec:
         securityContext:
           {{- toYaml . | nindent 8 }}
         {{- end }}
-        image: "{{ .Values.agent.image.name }}:{{ .Values.agent.image.tag |  default .Values.prefectVersionTag }}"
+        {{- /* We need to trim `core-` off the prefectVersionTag to include the K8s orchestration extras */}}
+        image: "{{ .Values.agent.image.name }}:{{ .Values.agent.image.tag |  default (.Values.prefectVersionTag | replace "core-" "") }}"
```

Comment:
Why not set the default image name in the chart values  to the latest core release (instead of `null`) when releasing the helm chart (since we already pull the core version as part of the release process)? May be cleaner.

Reply:
I can't think of a simple way to target the agent image with `sed`. Perhaps worth revisiting if we add a Python build-tool into the mix that can just load the yaml.

Reply:
Actually now that we're not using `core-X` to pull a UI version we can just set the `prefectVersionTag` without `core`

Reply:
Although I may want to strip the `core-` here anyway? I'm not sure what's best here

Reply:
If a user provides something that's invalid, we should error, not implicitly assume we know what they want. What if it's a non-prefect image that starts with `core-`?